lua/luasnip/loaders/from_vscode.lua

```diff
@@ -187,12 +187,42 @@ function M.load(opts)
 		opts.paths = get_snippets_rtp()
 	end
 
+	opts.paths = vim.tbl_map(expand_path, opts.paths) -- Expand before deduping, fake paths will become nil
+	opts.paths = vim.tbl_keys(list_to_set(opts.paths)) -- Remove doppelgänger paths and ditch nil ones
+
 	for _, path in ipairs(opts.paths) do
-		local full_path = expand_path(path)
-		if full_path then
-			load_snippet_folder(full_path, opts)
+		load_snippet_folder(path, opts)
+	end
+end
+
+local lazy_load_paths = {}
+local lazy_loaded_ft = {}
+
+function _G._luasnip_vscode_lazy_load()
+	for _, ft in ipairs({ vim.bo.filetype, "all" }) do
+		if not lazy_loaded_ft[ft] then
+			lazy_loaded_ft[ft] = true
+			M.load({ paths = lazy_load_paths, include = { ft } })
 		end
 	end
 end
 
+function M.lazy_load(opts)
+	opts = opts or {}
+
+	-- We have to do this here too, because we have to store them in lozy_load_paths
+	if type(opts.paths) ~= "table" and opts.paths ~= nil then
+		opts.paths = vim.split(opts.paths, ",")
+	else
+		opts.paths = get_snippets_rtp()
+	end
```

Comment:
The paths should also be filtered here, should they not?

Reply:
Here is not that important, the only issue could be that the user calls the function too many times (like in a loop) cause that will make the cache of paths grow. But yes, let me at least dedupe them.

Reply:
Ah ofc, completely overlooked that `M.load` is called on each path.